rtgui/edit.h

```diff
@@ -152,6 +145,59 @@ class EditBuffer;
  *
  */
 
+
+
+class ObjectMOBuffer
+{
+private:
+
+    // Used to draw the objects where the color correspond to the object's ID, in order to find the correct object when hovering
+    Cairo::RefPtr<Cairo::ImageSurface> objectMap;
+    // If more than 254 objects has to be handled, objectMap2 contains the "upper part" of the 16 bit int value. objectMap2 will be NULL otherwise.
+    Cairo::RefPtr<Cairo::ImageSurface> objectMap2;
+    ObjectMode objectMode;
+
+protected:
+
+    // To avoid duplicated information, we points to a EditDataProvider that contains the current EditSubscriber
+    // instead of pointing to the EditSubscriber directly
+    EditDataProvider* dataProvider;
+
+    void createBuffer(int width, int height);
+    void resize(int newWidth, int newHeight, EditSubscriber* newSubscriber);
+    void flush();
+    EditSubscriber *getEditSubscriber ();
+
+public:
+    ObjectMOBuffer(EditDataProvider *dataProvider);
+    ~ObjectMOBuffer();
+
+    EditDataProvider* getDataProvider()
```

Comment:
Personally, I find it preferable to move even inline method definitions outside of the class body (and add the `inline` modifier) so that a class contains only declarations a one can have a better overview. (Most IDE offer that as an automatic refactoring nowadays. Move definition out of class body or something like it.)